neurobazaar/services/datastorage/datastore_manager.py

```diff
@@ -0,0 +1,120 @@
+from urllib import request
+from uuid import UUID
+
+from django.shortcuts import render
+from home.models import LocalFSDatastores, MongoDBDatastores
+from neurobazaar.services.datastorage.abstract_datastore import AbstractDatastore
+from neurobazaar.services.datastorage.localfs_datastore import LocalFSDatastore
+from neurobazaar.services.datastorage.mongodb_datastore import MongoDBDatastore
+from django.db.models.deletion import ProtectedError
+
+import threading
+import functools
+
+datastore_manager = None
+
+def getDataStoreManager():
+    global datastore_manager
+    if datastore_manager is None:
+        datastore_manager = DatastoreManager()
+        datastore_manager.refresh()
+    return datastore_manager
+
+def synchronized(func):
+    lock = threading.Lock()
+    
+    def wrapper(*args, **kwargs):
+        with lock:
+            return func(*args, **kwargs)
+    
+    return wrapper
+
+class DatastoreManager:
+    def __init__(self):
+        self._datastores = {}
+    
+    @synchronized
+    def refresh(self):
+        print("Refreshing datastores...")
+        self.refreshLocalFSDatastores()
+        self.refreshMongoDBDatastores()
+        print(f"Datastores after refresh: {self._datastores}")
+    
+    def refreshLocalFSDatastores(self):
+        records = {record.UUID: record for record in LocalFSDatastores.objects.all()}
+        print(f"LocalFSDatastores records: {records}")
```

Comment:
This should be just a debugging message and is not supposed to print all the time.